### DISCUSSION TITLE
Pagination in PrimveVue Datatable not showing the correct page links

### DISCUSSION THREAD

**Discussion by sajjad201** [primefaces/community#242] [PrimeVue]
I am using pagination in primevue datatable as it: 
`<DataTable ref="dt" :value="dataItems" dataKey="id" :loading="loading1" :paginator="true" :totalRecords="50" :rows="5" :pageLinkSize="50" @page="handlePagination">
`
**What its Doing**
It shows no link to next/prev pages if server returns 5 data rows. 
if server return 10 rows, then it is showing 2 page links in pagination bar and so on... 

**What it should do**
There should be page links based on total count. f
For example, if server return totalRows=50 then it should show 10 page links in pagination bar and 5 data rows on each page so If user click on page 2, 3, 4... the request should go to server to fetch the related page data. 

Event Data on clicking the page link
`{
    "originalEvent": {
        "page": 1,
        "first": 4,
        "rows": 4,
        "pageCount": 2
    },
    "first": 4,
    "rows": 4,
    "sortField": null,
    "sortOrder": null,
    "multiSortMeta": [],
    "filters": {},
    "pageCount": 2,
    "page": 1
}` 

What changes should I made to generate links according to totalRecords so I can send request to server to get other pages data as well. 
Is it possible with https://primevue.org/paginator

**Comment by tugcekucukoglu** [comment]
> It shows no link to next/prev pages if server returns 5 data rows.

I think I don't understand what you mean. Is totalRecords = 5 or rows = 5?

**Reply by sajjad201** [reply]
Thanks for your input however I got the answer I actually was in search of lazy loading of Datatable. 
Lazy loading gets the data rows and totalRecords from server via api and then make total number of (pagination) links according to totalRecords which I was trying to achieve. 
By Adding 'lazy' in <Datatable lazy> solved my problem.

**Comment by Neoniq** [comment]
Same problem here (not showing pages on :totalRecords='111' set. Resolved by adding lazy as @sajjad201 points out
Suggest to add info in docs that `totalRecords` not working without `lazy`

https://stackblitz.com/edit/cryfmj?file=src%2FApp.vue

Created PR
https://github.com/primefaces/primevue/pull/4683

**Reply by Neoniq** [reply]
#4683
